google-cloud-asset/samples/acceptance/quickstart_test.rb

```diff
@@ -46,9 +46,19 @@
     sleep 3
     asset_name_list
   end
+  let :dataset do
+    create_dataset_helper "ruby_asset_sample_#{SecureRandom.hex}"
+  end
+  let :dataset_name do
+    dataset_name = "/datasets/#{dataset.dataset_id}"
+    # ensure read_time_window is after dataset creation
+    sleep 3
+    dataset_name
+  end
 
   after do
     delete_bucket_helper bucket.name
+    delete_dataset_helper dataset.dataset_id
```

Comment:
This will cause a dataset to be created (and immediately deleted) for _every_ quickstart tests, not just the ones that need it, because you're calling `dataset` (which performs the creation) here. Instead, please include some logic to determine whether `dataset` has actually been called by this test (e.g. by setting a flag in the `let :dataset` block, and testing that flag before deleting).

Reply:
Moved the delete call to the test that involves dataset.